From e2fd05457cc2279e8b7b8eb0ed1d334be978c795 Mon Sep 17 00:00:00 2001
From: Yong Li <yong.li@intel.com>
Date: Wed, 28 Feb 2024 13:29:05 +0800
Subject: [PATCH 54/63] lib: sbi: Add field to allow variety of rpxy protocl

The SBI RPXY extension proposal focused only on RPMI
protocol but it does make sense to allow variety of
messaging protocols (e.g. RPMI, secure, custom, etc.).

Signed-off-by: Yong Li <yong.li@intel.com>
---
 include/sbi/sbi_rpxy.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/sbi/sbi_rpxy.h b/include/sbi/sbi_rpxy.h
index f52e802..5f1a238 100644
--- a/include/sbi/sbi_rpxy.h
+++ b/include/sbi/sbi_rpxy.h
@@ -12,6 +12,15 @@
 
 #include <sbi/sbi_list.h>
 
+/** RPXY transport protocol type */
+#define RPXY_TRANS_PROT_MASK	(0xFFFF0000)
+#define RPXY_TRANS_PROT_SHIFT	(16)
+enum rpxy_transport_protocol {
+	RPXY_TRANS_PROT_RPMI = 0,
+	RPXY_TRANS_PROT_SEC = 1,
+	RPXY_TRANS_PROT_ID_MAX_COUNT,
+};
+
 struct sbi_scratch;
 
 /** A RPMI proxy service accessible through SBI interface */
-- 
2.34.1

